src/services/dumpers/agent-nodejs-dumper.ts

```diff
@@ -0,0 +1,176 @@
+import type { ConfigInterface, DbConfigInterface } from '../../interfaces/project-create-interface';
+
+import _ from 'lodash';
+
+import toValidPackageName from '../../utils/to-valid-package-name';
+import AbstractDumper from './abstract-dumper';
+
+export default class AgentNodeJsDumper extends AbstractDumper {
+  private env: { FOREST_SERVER_URL: string; FOREST_URL_IS_DEFAULT: string };
+
+  private readonly DEFAULT_PORT = 3310;
+
+  private readonly isLinuxOs: boolean;
+
+  constructor(context) {
+    const { assertPresent, env, isLinuxOs } = context;
+
+    assertPresent({
+      env,
+      isLinuxOs,
+    });
+
+    super(context);
+
+    this.env = env;
+    this.isLinuxOs = isLinuxOs;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  protected get templateFolder() {
+    return 'agent-nodejs/';
+  }
+
+  writePackageJson(dbDialect: string, applicationName: string) {
+    const dependencies: { [name: string]: string } = {
+      dotenv: '^16.0.1',
+      '@forestadmin/agent': '^1.0.0',
+    };
+
+    if (dbDialect === 'mongodb') {
+      dependencies['@forestadmin/datasource-mongoose'] = '^1.0.0';
+    } else {
+      dependencies['@forestadmin/datasource-sql'] = '^1.0.0';
+    }
+
+    if (dbDialect) {
+      if (dbDialect.includes('postgres')) {
+        dependencies.pg = '~8.2.2';
+      } else if (dbDialect === 'mysql') {
+        dependencies.mysql2 = '~2.2.5';
+      } else if (dbDialect === 'mssql') {
+        dependencies.tedious = '^6.4.0';
+      }
+    }
+
+    const pkg = {
+      name: toValidPackageName(applicationName),
+      version: '0.0.1',
+      private: true,
+      main: 'index.js',
+      scripts: {
+        'start:watch': 'nodemon',
+        start: 'node ./index.js',
+      },
+      nodemonConfig: {
+        ignore: ['./forestadmin-schema.json'],
+      },
+      dependencies,
+      devDependencies: {
+        nodemon: '^2.0.12',
+      },
+    };
+
+    this.writeFile('package.json', `${JSON.stringify(pkg, null, 2)}\n`);
+  }
+
+  writeIndex(dbDialect: string) {
+    const isMongoose = dbDialect === 'mongodb';
+
+    const context = {
+      isMongoose,
+      isMySQL: dbDialect === 'mysql',
+      isMSSQL: dbDialect === 'mssql',
+      forestServerUrl: !!this.env.FOREST_SERVER_URL,
+      datasourceImport: null,
+      datasourceCreation: isMongoose
+        ? 'createMongooseDataSource(models.connections.default, {}), {});'
+        : `createSqlDataSource({
+      uri: process.env.DATABASE_URL,
+      schema: process.env.DATABASE_SCHEMA || 'public',
+      ...dialectOptions,
+    }),`,
+    };
+
+    this.copyHandleBarsTemplate('index.hbs', 'index.js', context);
+  }
+
+  private writeDotEnv(
+    dbConfig: DbConfigInterface,
+    applicationPort: number,
+    forestEnvSecret: string,
+    forestAuthSecret: string,
+  ) {
+    const databaseUrl = this.buildDatabaseUrl(dbConfig);
+    const context = {
+      databaseUrl,
+      databaseSsl: dbConfig.ssl || false,
```

Comment:
I don't understand the switch here from dbSsl to databaseSsl. Sometimes it database sometimes it is db, we need to commit on a pattern.

Reply:
I agree, the `ssl` in the dbConfig should be renamed too, we have so many disparities...
I'll fix this here.